name: DaCe RTL hardware emulation
on: workflow_dispatch
jobs:
  test-rtl:
    runs-on: [self-hosted, linux, xilinx-fpga]
    steps:
    - uses: actions/checkout@v4
      with:
        submodules: 'recursive'
    - name: Install dependencies
      run: |
        rm -f ~/.dace.conf
        rm -rf .dacecache tests/.dacecache
        . /opt/setupenv
        python -m pip install --upgrade pip
        pip install pytest-xdist flake8
        pip uninstall -y dace
        pip install -e ".[testing]"

    - name: Run FPGA Tests
      run: |
        # Due to an internal bug in the Xilinx tools, where the current datetime is passed as an integer
        # and overflowed in the year 2022, run the RTL FPGA tests pretending like it's January 1st 2021.
        faketime -f "@2021-01-01 00:00:00" pytest -n auto --tb=short -m "rtl_hardware"
        killall -9 xsim xsimk || true